Singularity/Singularity/Screen/Slider.cs

```diff
@@ -157,63 +176,264 @@ public void Update(GameTime gametime)
+                        Color.Gray,
+                        Color.Black,
+                        1,
+                        (float)0.8);
+
+                    if (!mWithPages)
+                    {
+                        // draws in value of slider in the center of display window
+                        spriteBatch.DrawString(mFont,
+                            origin: Vector2.Zero,
+                            position: new Vector2((mMax + Size.Y + 30) - (mFont.MeasureString(mStringValue).X / 2),
+                                Position.Y - 12),
+                            color: Color.White,
+                            text: mStringValue.ToString(),
+                            rotation: 0f,
+                            scale: 1f,
+                            effects: SpriteEffects.None,
+                            layerDepth: 0.2f);
+                    }
+                    else
+                    {
+                        if (mPageSize > 3)
+                        {
+                            // draw page notches if they arent to close to another on bar
+                            for (int i = 0; i < (Pages+1); i++)
+                            {
+                                spriteBatch.DrawLine(new Vector2((Position.X + (i * (Size.X / Pages))), Position.Y - 2),
+                                    4,
+                                    1.57f,
+                                    Color.White * .8f,
+                                    1);
+                            }
+                        }
+
+                        // draws in page number of slider in the center of display window
+                        spriteBatch.DrawString(mFont,
+                            origin: Vector2.Zero,
+                            position: new Vector2((mMax + Size.Y + 30) - (mFont.MeasureString(mCurrentPage.ToString()).X / 2),
+                                Position.Y - 12),
+                            color: Color.White,
+                            text: mCurrentPage.ToString(),
+                            rotation: 0f,
+                            scale: 1f,
+                            effects: SpriteEffects.None,
+                            layerDepth: 0.2f);
+
+                    }
+                
+                 // draws slider bar
+                 spriteBatch.DrawLine(Position, Size.X, 0f, Color.White*0.6f, 2);
+
+                // draws slider
+                    spriteBatch.StrokedRectangle(
+                        new Vector2(mCurrentX - (Size.Y / 2), Position.Y - (Size.Y / 2)),
+                        new Vector2(Size.Y, Size.Y),
+                        Color.Gray,
+                        Color.Black,
+                        (float) .5,
+                        (float) 0.8);
 
                 }
             }
         }
 
+        #region MouseActionsWhichDontWork
+
+        // TODO not woriking, wont print debug message
```

Comment:
Why is it there though it's not working? (is this still up-to-date?)

Reply:
so i have the mouse actions in the update method right now because the input manager stuff isnt working for some reason. no clue why. but leon needs the stuff for his tasks this week, so i figured we can figure that out at some point later, that shouldn't change anything